### PR TITLE
Add API for indicating large data scenario in ModelAssessmentContext

### PR DESCRIPTION
**Pull request by gaugup**
## Description

Adding API `ifEnableLargeData`  to indicate and serve the big data scenario.

## Checklist

- [ ] I have added screenshots above for all UI changes.
- [ ] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

File: libs/core-ui/src/lib/util/buildInitialContext.ts [hunk 1]
```diff
@@ -125,3 +126,10 @@ export function getModelType(
       return ModelTypes.Multiclass;
   }
 }
+
+export function ifEnableLargeData(dataset: IDataset): boolean {
+  if (dataset?.is_large_data_scenario && dataset?.use_entire_test_data) {
```
Comment (hawestra) [inline libs/core-ui/src/lib/util/buildInitialContext.ts:131]: nit: just return dataset?.is_large_data_scenario && dataset?.use_entire_test_data

**Review by hawestra — APPROVED** [review]

File: libs/core-ui/src/lib/Interfaces/IDataset.ts [hunk 2]
```diff
@@ -19,6 +19,8 @@ export interface IDataset {
   features: unknown[][];
   feature_names: string[];
   categorical_features: string[];
+  is_large_data_scenario?: boolean;
```
Comment (imatiach-msft) [inline libs/core-ui/src/lib/Interfaces/IDataset.ts:22]: I think you need to whitelist the names to make the linter pass here:

https://github.com/microsoft/responsible-ai-toolbox/blob/2bd09f98fdffe3e18c6a04f92b127e727da747c9/.eslintrc/.eslintrc.custom.eslintrc

**Review by imatiach-msft — APPROVED** [review]

**Review by vinuthakaranth — APPROVED** [review]

File: libs/core-ui/src/lib/Interfaces/IDataset.ts (hunk 2 above)
Comment (gaugup) [reply libs/core-ui/src/lib/Interfaces/IDataset.ts:22]: Done. Thanks for pointing this out.

File: libs/core-ui/src/lib/util/buildInitialContext.ts (hunk 1 above)
Comment (gaugup) [reply libs/core-ui/src/lib/util/buildInitialContext.ts:131]: Looks like returning the ```dataset?.is_large_data_scenario && dataset?.use_entire_test_data;``` raises an error.

```
ERROR in libs/core-ui/src/lib/util/buildInitialContext.ts:131:3
TS2322: Type 'boolean | undefined' is not assignable to type 'boolean'.
  Type 'undefined' is not assignable to type 'boolean'.
    129 |
    130 | export function ifEnableLargeData(dataset: IDataset): boolean {
  > 131 |   return dataset?.is_large_data_scenario && dataset?.use_entire_test_data;
        |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    132 | }
    133 |
```

Comment (gaugup) [reply libs/core-ui/src/lib/util/buildInitialContext.ts:131]: Is there any other way to simplify this code?

**Pull request merged**